### PR TITLE
Digging

### PR DESCRIPTION
**Pull request by Vordenburg**
No longer will Glacier and Lighthouse have terrible plumbing.

Most of this code is cribbed from a mix of CutLattice and PryTile. I added some tile items where necessary. The digging flow goes from Snow/Grass[dark/light/jungle] -> Dirt -> Bedrock. Bedrock is an entirely new tile I made, so plain dirt can be used to hide subfloor entities. For the record: snow, any grass variant, and dirt will hide subfloor entities, whereas bedrock will not. Bedrock is like the planetoid variant of plating.

I haven't addressed the silliness of things like "tiles of snow" in this PR. Maybe another time.

:cl:
- add: Shovels can now be used to dig up earthen tiles.
- fix: Dirt, grass, snow, and other earthen tiles will hide wires and pipes again.

### CODE REVIEW

File: Content.Server/Tools/ToolSystem.EarthDigging.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+using Content.Shared.Tools.Components;
+
+namespace Content.Server.Tools;
+
+public sealed partial class ToolSystem
+{
+    private void InitializeEarthDigging()
+    {
+        SubscribeLocalEvent<EarthDiggingComponent, AfterInteractEvent>(OnEarthDiggingAfterInteract);
+        SubscribeLocalEvent<EarthDiggingComponent, EarthDiggingCompleteEvent>(OnEarthDigComplete);
```
Comment (Elijahrane) [inline Content.Server/Tools/ToolSystem.EarthDigging.cs]: For doafter stuff like these it's usually

```SubscribeLocalEvent<EarthDiggingCompleteEvent>```

and then you TryComp for the component in the subscription to it

This way prevents some potential bugs that can happen if the component is changed/added/removed between the doafter starting and finishing or getting cancelled.

**Review by Elijahrane — CHANGES_REQUESTED** [review]

File: Content.Server/Tools/ToolSystem.EarthDigging.cs (hunk 1 above)
Comment (Vordenburg) [reply Content.Server/Tools/ToolSystem.EarthDigging.cs]: TilePrying did it this way, so I just followed convention. I can see how that could create a problem though. I would've thought the event wouldn't fire if the component reached those lifestages though.

Comment (Elijahrane) [reply Content.Server/Tools/ToolSystem.EarthDigging.cs]: tilepry is really old, what i posted is the current convention

Comment (Vordenburg) [reply Content.Server/Tools/ToolSystem.EarthDigging.cs]: I figured it out. It was necessary to remove the `doAfterEventTarget` parameter being supplied to UseTool in order to get it to fire a broadcast event as opposed to a directed event.

**Pull request merged**